### PR TITLE
Actions table

### LINKED ISSUE #217: Move actions into a separate SQL table
Move Actions into their own table from the Transactions table

### PR DESCRIPTION
**Pull request by icerove**
fix #217

### CODE REVIEW

File: backend/src/sync.js [hunk 1]
```diff
@@ -52,29 +52,44 @@ async function saveBlocks(blocksInfo) {
...
                       signerId: tx.signer_id,
                       signerPublicKey: tx.signer_public_key || tx.public_key,
                       signature: tx.signature,
-                      receiverId: tx.receiver_id,
-                      actions
+                      receiverId: tx.receiver_id
                     };
                   })
                 ),
+                blockInfo.transactions.map(tx => {
```
Comment (frol) [inline backend/src/sync.js]: JS fails you here. `Promise.all()` accepts a list of promises, and so we have a list of:

* one promise created with `models.Transaction.bulkCreate`
* a list of promises created for each `transaction` in `blockInfo.transactions` with `models.Action.bulkCreate`
* one promise created with `models.Account.bulkCreate`

`Promise.all` will consider the list of `models.Action.bulkCreate` promises as an already resolved value, and will not await for it.

Concatenate the list of `models.Transaction.bulkCreate` and `models.Account.bulkCreate` promises with the list of promises created with the `blockInfo.transactions.map`.

Example:

```js
async function qq1() {
  const x = Promise.all(
    [
      [new Promise((resolve) => setTimeout(() => { console.log('QQ1: resolved'); resolve() }, 1000))]
    ]
  );
  await x;
  console.log('QQ1: all the promises are resolved');
};

async function qq2() {
  const x = Promise.all(
    [
      new Promise((resolve) => setTimeout(() => { console.log('QQ2: resolved'); resolve() }, 1000))
    ]
  );
  await x;
  console.log('QQ2: all the promises are resolved');
};
```

When you run `qq1`, you will get:

```
QQ1: all the promises are resolved
QQ1: resolved
```

That is not what you want.

When you run `qq2`, you will get:

```
QQ2: resolved
QQ2: all the promises are resolved
```

File: frontend/src/libraries/explorer-wamp/transactions.ts [hunk 2]
```diff
@@ -172,16 +172,22 @@ export default class TransactionsApi extends ExplorerApi {
           //
           // Once the above TODO is resolved, we should just move this to TransactionInfo method
           // (i.e. query the information there only for the specific transaction).
-          const actions = transactionExtraInfo.transaction.actions;
-
-          transaction.actions = actions.map((action: RpcAction | string) => {
+          const actions = await this.call<any>("select", [
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/transactions.ts]: Well, given the above comments, we cannot do this yet :disappointed: 

You may notice that we store Actions in the DB, but never use them, and you are absolutely right. this is the case now.

Let's keep this code as a comment here, so in future we can simply uncomment it.

Comment (icerove) [reply frontend/src/libraries/explorer-wamp/transactions.ts]: okaaaaay

File: backend/src/sync.js (hunk 1 above)
Comment (frol) [reply backend/src/sync.js]: @icerove Please, familiarize yourself with Promises and async/await in JS. Here is what you essentialy have here now:

```
Promise.all(
  [
     [undefined, undefined, ...],
     Promise-to-bulk-create-Accounts
  ]
)
```

Well, we should also refactor this code (break it apart).

File: frontend/src/libraries/explorer-wamp/transactions.ts (hunk 2 above)
Comment (frol) [reply frontend/src/libraries/explorer-wamp/transactions.ts]: Well, Wallet does use actions available from the Explorer Backend, so we should also have a PR for Wallet.

**Pull request merged**